### PR TITLE
Some WebGL drive-by fixes

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21207)

### CODE REVIEW

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -3612,7 +3615,12 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
     ) -> JSVal {
         // FIXME(nox): https://github.com/servo/servo/issues/21133
 
-        if program.is_deleted() || !program.is_linked() || program.id() != location.program_id() {
+        if
+            program.is_deleted() ||
+            !program.is_linked() ||
+            program.id() != location.program_id() ||
```
Comment (emilio) [inline components/script/dom/webglrenderingcontext.rs]: Maybe worth moving this into a `UniformLocation::linked_in(&WebGLProgram)` or something like that?

**Review by emilio — APPROVED** [review]
Very nice :)

File: components/script/dom/webglrenderingcontext.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/webglrenderingcontext.rs]: Yeah ultimately I want to refactor all the things but I'm not sure where it should go and how it should like for now.

**Comment by nox** [comment]
Last commit makes that test fail, probably because we lack some bookkeeping somewhere in DOM. Removing for now.

**Comment by jdm** [comment]
@bors-servo retry
* Websockets

**Comment by jdm** [comment]
@bors-servo retry
* #21212

**Pull request merged**